numpy/core/memmap.py

```diff
@@ -309,3 +309,20 @@ def flush(self):
         """
         if self.base is not None and hasattr(self.base, 'flush'):
             self.base.flush()
+
+    def __array_wrap__(self, arr, context=None):
+        # Return memmap if self and arr are likely to share memory
```

Comment:
p.s. Looking at #2932, I think it would be nice to properly account for possibly being not directly an `ndarray` subclass (but, e.g., a mixed type like `class(Quantity, memmap)`, i.e., I think I would here include the following
```
    arr = super(memmap, self).__array_wrap__(arr, *args, **kwargs)
    # Return arr if it not a memmap or if it was given as output.
    if arr is self or type(arr) is not memmap:
        return arr
    # If we were created in the `ufunc`, we should not be a memmap,
    # but return a scalar or ndarray, as appropriate.
    <your lines>
```